packages/camera/camera_android_camerax/lib/src/android_camera_camerax.dart

```diff
@@ -447,11 +499,14 @@ class AndroidCameraCameraX extends CameraPlatform {
           height: imageProxy.height,
           width: imageProxy.width);
 
-      cameraImageDataStreamController?.add(cameraImageData);
+      weakThis.target!.cameraImageDataStreamController!.add(cameraImageData);
```

Comment:
First issue: @bparrishMines am I misunderstanding how weakThis should be used? Should I allow for the target to be null to avoid leaks?

Second issue: Yes, this is a helper method used to set up the image streaming that is called from listening to the stream returned by onStreamedFrameAvailable, so it should never be called really any other time. cameraImageDataStreamController is set up before that, so it would never be null here.

Reply:
I think the point of the weak reference is to ensure the callback method doesn't prevent garbage collection of the instance that contains it. The `target` COULD be null, but the `!` is just a runtime check that can be used if you expect the `target` to always be available in the callback.

For example, if you expect the user to eventually ALWAYS call `CameraController.dispose` and you do some kind of cleanup in that method that stops image streaming callbacks. Then I think it is reasonable to expect this to always be nonnull. But a test would also be helpful for ensuring this and preventing a null pointer exception.

As a comparison, `weview_flutter` always uses `weakThis.target?...` because there is no `WebViewController.dispose`.